### PR TITLE
Add infrastructure proposal to discuss choosing an out-of-process only design for hosting the hlsl compiler api

### PR DESCRIPTION
**Pull request by coopp**
This commit adds a proposal to the infrastructure proposals to build an out-of-process only design to hose the hlsl compiler api. This proposal will go under a design discussion which could update/change it.

### CODE REVIEW

File: proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<!-- {% raw %} -->
+
+# Support for in-process, Out-of-process, or both for the c style compiler api
+
+* Proposal: [NNNN](INF-NNNN-inproc-outofproc-compiler-api-support.md)
```
Comment (coopp) [inline proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md:5]: It was mentioned in a previous PR that the name of this proposal doesn't fully match the contents.  Originally this was a discussion type proposal but it reads like a decision has been made to choose out of process.  There will be a design meeting that may change the contents here, so I think I can keep this name.  Any objections to this?

Comment (damyanp) [reply proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md:5]: All proposals in this repo are "infra" ones from hlsl specs, so there's no need for the infra prefix or for adding the infra directory.

Comment (coopp) [reply proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md:5]: I'll fix this up.  I wasn't sure.

Comment (coopp) [reply proposals/infra/INF-NNNN-inproc-outofproc-compiler-library-support.md]: fixed!

**Review by damyanp — APPROVED** [review]
We should try and figure out when it is appropriate to assign a number to these proposals.

**Review by dmpots — APPROVED** [review]

**Review by bogner — APPROVED** [review]

**Pull request merged**